src/workers/cdc/primary.ts

```diff
@@ -857,6 +858,37 @@ const onSquadPublicRequestChange = async (
   });
 };
 
+const WEEKEND_MAP: Partial<Record<DayOfWeek, DayOfWeek[]>> = {
+  [DayOfWeek.Sunday]: [5, 6],
+  [DayOfWeek.Monday]: [6, 0],
+};
+
+const getNextWeekday = (todayTz: Date, weekStart: DayOfWeek): Date => {
+  const weekends = WEEKEND_MAP[weekStart];
+
+  if (!weekends) {
+    throw new Error('Invalid week start: ' + weekStart);
+  }
+
+  if (!weekends.includes(todayTz.getDay())) {
+    return addDays(todayTz, 1);
+  }
+
+  const monday = nextMonday(todayTz);
+  const weekday = weekStart === DayOfWeek.Sunday ? monday : addDays(monday, 1);
```

Comment:
> We can maybe use nextMonday and nextSunday instead?

Good point, we can stick to always using functions from date-fns.

> Should it not be reverse if weekStart is Sunday then we need next Sunday not Monday?

We want the EOD of the start of the week, so that means it would be Monday 00:00:00 when your start of week is Sunday

Reply:
> We want the EOD of the start of the week, so that means it would be Monday 00:00:00 when your start of week is Sunday

Yeah, this was confusing from variable naming, maybe specific calls would make it more logical.

Reply:
Right. Let me refactor to make it clearer.